src/Microsoft.Azure.ServiceBus/Core/MessageReceiver.cs

```diff
@@ -1194,6 +1241,39 @@ static void CloseRequestResponseSession(RequestResponseAmqpLink requestResponseA
                 amqpRequestMessage.Map[ManagementConstants.Properties.LockTokens] = lockTokens;
                 amqpRequestMessage.Map[ManagementConstants.Properties.DispositionStatus] = dispositionStatus.ToString().ToLowerInvariant();
 
+                if (deadLetterReason != null)
+                {
+                    amqpRequestMessage.Map[ManagementConstants.Properties.DeadLetterReason] = deadLetterReason;
+                }
+
+                if (deadLetterDescription != null)
+                {
+                    amqpRequestMessage.Map[ManagementConstants.Properties.DeadLetterDescription] = deadLetterDescription;
+                }
+
+                if (propertiesToModify != null)
+                {
+                    var amqpPropertiesToModify = new AmqpMap();
+                    foreach (var pair in propertiesToModify)
+                    {
+                        object amqpObject;
+                        if (AmqpMessageConverter.TryGetAmqpObjectFromNetObject(pair.Value, MappingType.ApplicationProperty, out amqpObject))
+                        {
+                            amqpPropertiesToModify[new MapKey(pair.Key)] = amqpObject;
+                        }
+                    }
+
+                    if (amqpPropertiesToModify.Count > 0)
```

Comment:
Looking at the [logic above](https://github.com/Azure/azure-service-bus-dotnet/pull/341/files?diff=unified#diff-f0c668c8659c9b5c15eb016bcd714028R1260) there's a chance that not all properties could end up in the `amqpPropertiesToModify` collection. In that case, should there be a check and a warning logged if it happens?